### PR TITLE
ADD keyboard Vxe75

### PR DESCRIPTION
**Pull request by gskygithub**
## Description

Add vxe75 keyboard

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/vxe75/config.h [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+#define DRIVER_2_EN C4
+
+#define DRIVER_COUNT 2
+#define DRIVER_1_LED_TOTAL 65
+#define DRIVER_2_LED_TOTAL 15
+#define RGB_MATRIX_LED_COUNT (DRIVER_1_LED_TOTAL + DRIVER_2_LED_TOTAL)
+
+#define RGB_DISABLE_WHEN_USB_SUSPENDED
+#define RGB_MATRIX_FRAMEBUFFER_EFFECTS
+#define RGB_MATRIX_KEYPRESSES
```
Comment (zvecr) [inline keyboards/vxe75/config.h:35]:
```suggestion

```

File: keyboards/vxe75/config.h [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+/* Flash */
+#define EXTERNAL_FLASH_SPI_SLAVE_SELECT_PIN B14
+
+/* LED */
+#define LED_PIN_ON_STATE 0
+#define LED_SCROLL_LOCK_PIN C0
+#define LED_WIN_LOCK_PIN C1
+#define LED_CAPS_LOCK_PIN C2
+#define LED_NUM_LOCK_PIN C3
+
```
Comment (zvecr) [inline keyboards/vxe75/config.h:21]:
```suggestion
```

File: keyboards/vxe75/info.json [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+        "wear_leveling": {
+            "backing_size": 4096,
+            "driver": "spi_flash"
+        }
+    },
+    "encoder": {
+        "rotary": [
+            {"pin_a": "C14", "pin_b": "C15"}
+        ]
+    },
```
Comment (zvecr) [inline keyboards/vxe75/info.json:40]:
```suggestion
    },
    "indicators": {
        "caps_lock": "C2",
        "num_lock": "C3",
        "scroll_lock": "C0",
        "on_state": 0
    },
```

**Review by zvecr — COMMENTED** [review]
Also, as flagged by lint:
* no default keymap
* bad line endings

File: keyboards/vxe75/info.json [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+    "rgb_matrix": {
+        "driver": "aw20216",
+        "max_brightness": 108,
+        "val_steps": 27,
+        "speed_steps": 64,
```
Comment (zvecr) [inline keyboards/vxe75/info.json:45]: Poor choice of defaults
```suggestion
```

File: keyboards/vxe75/info.json [hunk 5]
```diff
@@ -0,0 +1,266 @@
...
+        "animations": {
+            "solid_color": true,
```
Comment (zvecr) [inline keyboards/vxe75/info.json:47]:
```suggestion
```

File: keyboards/vxe75/keymaps/via/keymap.c [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+    [_LMF] = LAYOUT(
+        _______,  KC_F1,    KC_F2,    KC_F3,    KC_F4,    KC_F5,    KC_F6,    KC_F7,    KC_F8,    KC_F9,    KC_F10,   KC_F11,   KC_F12,            _______,
+        RGB_HUI,  _______,  _______,  _______,  _______,   _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,
+        _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  KC_PSCR,  KC_SCRL,  KC_PAUS,  _______,  _______,  _______,  _______,
+        _______,  TO(_LB),  TO(_LM),  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,            _______,  RGB_MOD,
+        _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,            RGB_VAI,
+        _______,  GU_TOGG,  _______,                      RGB_TOG,                                _______,  _______,            RGB_SPD,  RGB_VAD,  RGB_SPI),
+};
+
+const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][2] = {
```
Comment (zvecr) [inline keyboards/vxe75/keymaps/via/keymap.c:48]:
```suggestion
const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][NUM_DIRECTIONS] = {
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/vxe75/readme.md [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+
+## Bootloader
+
+Enter the bootloader in 3 ways:
+
+* **Bootmagic reset**: Hold down the key at (0,0) in the matrix (usually the top left key or Escape) and plug in the keyboard
+* **Physical reset button**: Briefly press the button on the back of the PCB - some may have pads you must short instead
+* **Keycode in layout**: Press the key mapped to `QK_BOOT` if it is available
+
+See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
```
Comment (zvecr) [inline keyboards/vxe75/readme.md:22]:
```suggestion
See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).

## Bootloader

Enter the bootloader in 3 ways:

* **Bootmagic reset**: Hold down the key at (0,0) in the matrix (usually the top left key or Escape) and plug in the keyboard
* **Physical reset button**: Briefly press the button on the back of the PCB - some may have pads you must short instead
* **Keycode in layout**: Press the key mapped to `QK_BOOT` if it is available

```

File: keyboards/vxe75/vxe75.h
Comment (zvecr) [inline keyboards/vxe75/vxe75.h:1]: Still needs to be deleted.

**Review by zvecr — CHANGES_REQUESTED** [review]
* Multiple files still have line ending issues as flagged by lint

**Comment by gskygithub** [comment]
@zvecr Please take a look at this PR, thanks.

File: keyboards/vxe75/info.json
Comment (drashna) [inline keyboards/vxe75/info.json:1]: Please run `qmk format-json` on this file.

File: keyboards/vxe75/vxe75.c [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
+
+const aw_led g_aw_leds[RGB_MATRIX_LED_COUNT] = {
```
Comment (zvecr) [inline keyboards/vxe75/vxe75.c:19]:
```suggestion
const aw20216s_led_t PROGMEM g_aw20216s_leds[AW20216S_LED_COUNT] = {
```

File: keyboards/vxe75/config.h [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+#define SPI_MISO_PIN A6
+
+/* Flash */
+#define EXTERNAL_FLASH_SPI_SLAVE_SELECT_PIN B14
+
+/* RGB Driver */
+#define DRIVER_1_CS A15
+#define DRIVER_2_CS B15
+#define DRIVER_1_EN C4
+#define DRIVER_2_EN C4
```
Comment (zvecr) [inline keyboards/vxe75/config.h:19]:
```suggestion
#define AW20216S_CS_PIN_1 A15
#define AW20216S_CS_PIN_2 B15
#define AW20216S_EN_PIN C4
```

File: keyboards/vxe75/readme.md [hunk 10]
```diff
@@ -0,0 +1,23 @@
+# VXE75
+
+* Keyboard Maintainer:
```
Comment (zvecr) [inline keyboards/vxe75/readme.md:3]: Should not be blank?
```suggestion
* Keyboard Maintainer: Westberry
```

File: keyboards/vxe75/vxe75.c [hunk 11]
```diff
@@ -0,0 +1,112 @@
...
+/* Refer to IS31 manual for these locations
+ *   driver
+ *   |  R location
+ *   |  |          G location
+ *   |  |          |          B location
+ *   |  |          |          | */
+    {0, CS1_SW1,  CS2_SW1,  CS3_SW1   },  // 0
```
Comment (zvecr) [inline keyboards/vxe75/vxe75.c:26]: <https://docs.qmk.fm/#/feature_rgb_matrix?id=aw20216s>

Location names have been updated, for example
```
    { 0, SW1_CS1, SW1_CS2, SW1_CS3 },
    { 0, SW1_CS4, SW1_CS5, SW1_CS6 },
    { 0, SW1_CS7, SW1_CS8, SW1_CS9 },
```

These should be updated

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/vxe75/vxe75.h
Comment (drashna) [inline keyboards/vxe75/vxe75.h:1]: This file can and should be removed.

File: keyboards/vxe75/vxe75.c [hunk 12]
```diff
@@ -0,0 +1,112 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/vxe75/vxe75.c:17]:
```suggestion
#include "quantum.h"
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Comment by fauxpark** [comment]
Some formatting issues, otherwise looks OK.

File: keyboards/vxe75/info.json
Comment (fauxpark) [inline keyboards/vxe75/info.json:1]: This file should be renamed to `keyboard.json` and converted to Unix line endings.

File: keyboards/vxe75/rules.mk
Comment (fauxpark) [inline keyboards/vxe75/rules.mk:1]: This file can be removed.

File: keyboards/vxe75/readme.md
Comment (fauxpark) [inline keyboards/vxe75/readme.md:1]: Please convert this file to Unix line endings.

File: keyboards/vxe75/vxe75.c [hunk 13]
```diff
@@ -0,0 +1,112 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "quantum.h"
+
+const aw20216s_led_t PROGMEM g_aw20216s_leds[AW20216S_LED_COUNT] = {
```
Comment (fauxpark) [inline keyboards/vxe75/vxe75.c:19]:
```suggestion
#ifdef RGB_MATRIX_ENABLE
const aw20216s_led_t PROGMEM g_aw20216s_leds[AW20216S_LED_COUNT] = {
```

File: keyboards/vxe75/vxe75.c [hunk 14]
```diff
@@ -0,0 +1,112 @@
...
+    {0, SW1_CS16, SW1_CS17,  SW1_CS18 },  // 71
+    {0, SW2_CS16, SW2_CS17,  SW2_CS18 },  // 72
+    {0, SW3_CS16, SW3_CS17,  SW3_CS18 },  // 73
+    {0, SW6_CS16, SW6_CS17,  SW6_CS18 },  // 74
+    {0, SW9_CS16, SW9_CS17,  SW9_CS18 },  // 75
+    {0, SW10_CS16,SW10_CS17, SW10_CS18},  // 76
+    {1, SW10_CS4, SW10_CS5,  SW10_CS6 },  // 77
+    {1, SW11_CS4, SW11_CS5,  SW11_CS6 },  // 78
+    {1, SW5_CS10, SW5_CS11,  SW5_CS12 },  // 79
+};
```
Comment (fauxpark) [inline keyboards/vxe75/vxe75.c:112]:
```suggestion
};
#endif
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/vxe75/keyboard.json
Comment (fauxpark) [inline keyboards/vxe75/keyboard.json:1]: These two files still need to be switched to Unix line endings (LF). Right now they are Windows style (CRLF).

Please do not use the GitHub web editor for this, as it seems to ignore the `.gitattributes` file that enforces LF line endings.

**Pull request closed without merging**